src/domains/ecommerce-app/api/ecommerce-checkout/v1/_validate_transactions_jwt_token.tpl

```diff
@@ -3,8 +3,15 @@
         <base />
         <set-header name="x-transaction-id" exists-action="delete" />
         <set-variable name="requestTransactionId" value="@{
-            return context.Request.MatchedParameters["transactionId"];
-        }" />
+                    return context.Request.Headers.GetValueOrDefault("transactionId","");
```

Comment:
🚲 Instead of having a choose maybe the transactionId retrieval code can be all be written directly into this set-variable tag, something like:
var transactionId = context.Request.MatchedParameters.GetValueOrDefault("transactionId","");
if(transactionId == ""){
transactionId = context.Request.Headers.GetValueOrDefault("transactionId","");
}
return transactionId;

and remove the below choose tag altogether